tests/test_forms.py

```diff
@@ -170,6 +170,126 @@ def slot_mappings(self):
     assert slot_values == {"some_slot": "some_value"}
 
 
+def test_extract_requested_slot_from_entity_with_role():
+    """Test extraction of a slot value from entity with role."""
+
+    # noinspection PyAbstractClass
+    class CustomFormAction(FormAction):
+        def name(self):
+            return "some_form"
+
+        def slot_mappings(self):
+            return {
+                "some_slot": self.from_entity(entity="some_entity", role="some_role")
+            }
+
+    form = CustomFormAction()
+
+    tracker = Tracker(
+        "default",
+        {"requested_slot": "some_slot"},
+        {
+            "intent": {"name": "some_intent", "confidence": 1.0},
+            "entities": [
+                {
+                    "entity": "some_entity",
+                    "value": "some_value",
+                    "role": "some_other_role",
```

Comment:
Where if your other branch does this get added to the UserUttered event? (i took a short look but not too in depth)

Reply:
`UserUttered` has a list of entities (dict). Adding more values to the dict should be fine. They should be stored by default.